cypress/integration/unit_tests/click_feature_test.js

```diff
@@ -1,90 +1,201 @@
-    const feature = ee.Feature(null, featureProperties);
-    const featureCollection = ee.FeatureCollection([feature]);
-    cy.stub(ee, 'FeatureCollection')
-        .withArgs('mockAsset')
-        .returns(featureCollection);
-    cy.stub(featureCollection, 'filterBounds').returns(featureCollection);
-    cy.stub(featureCollection, 'first').returns(feature);
-    cy.stub(feature, 'evaluate').callsFake((callb) => callb({
-                                             'type': 'Feature',
-                                             'geometry':
-                                                 {coordinates: [[[0, 0]]]},
-                                             'properties': featureProperties,
-                                           }));
+  before(() => {
+    const feature1 = createFeatureFromCorners(...feature1Corners)
+                         .set(blockGroupTag, 'my block group');
+    const feature2 = createFeatureFromCorners(...feature2Corners)
+                         .set(blockGroupTag, 'another group');
+    const offMapFeature = createFeatureFromCorners(10, 10, 20, 20);
+    const otherFeature = createFeatureFromCorners(-11, -11, -10, -10);
+    features =
+        ee.FeatureCollection([feature1, feature2, offMapFeature, otherFeature]);
+    scoredFeatures = ee.FeatureCollection([
+      feature1.set(scoreTag, 1),
+      feature2.set(scoreTag, 3),
+      offMapFeature.set(scoreTag, 2),
+      otherFeature.set(scoreTag, 0),
+    ]);
   });
 
-  it('clicks on a block group in the list', () => {
-    const tableData = [tableHeadings, [0, 99, 0.46, 0.52]];
-    mockTable.expects('setSelection').once().withArgs([{row: 0, column: null}]);
+  beforeEach(() => currentFeatures.clear());
 
-    clickFeature(0, 0, null, 'mockAsset', tableApi, tableData);
+  /**
+   * Sets up map and table by calling real setup functions, but with our toy
+   * EE features.
+   * @return {Cypress.Chainable}
+   */
+  function setUpPage() {
+    const loadingFinishedPromise =
+        new Promise((resolve) => loading.loadingElementFinished = (id) => {
+          if (id === tableContainerId) resolve();
+        });
+    createGoogleMap().then((mapResult) => map = mapResult);
+    cy.document().then((doc) => {
+      // Lightly fake out prod document access.
+      cy.stub(document, 'getElementById')
+          .callsFake((id) => doc.getElementById(id));
+      cy.stub(document, 'createElement')
+          .callsFake((tag) => doc.createElement(tag));
+      const containerDiv = doc.createElement('div');
+      containerDiv.id = 'tableContainer';
+      doc.body.appendChild(containerDiv);
+      const tableDiv = doc.createElement('div');
+      tableDiv.id = 'table';
+      containerDiv.appendChild(tableDiv);
+      drawTableAndSetUpHandlers(
+          convertEeObjectToPromise(scoredFeatures).then((fc) => fc.features),
+          map, features);
+    });
+    cy.wrap(loadingFinishedPromise);
+    return assertNoSelection();
```

Comment:
why return if not being chained?

Reply:
someone might chain off of it. It's also a very good signal that this function does things in the "Cypress phase". The convention I'm using (described more explicitly in my follow-up PR) is that a function that does things in the Cypress phase returns a Chainable, while a function that does things in normal execution order returns a normal object, since why else would you be calling it.